aws-api/src/main/java/com/amplifyframework/api/aws/sigv4/AppSyncSigV4SignerInterceptor.java

```diff
@@ -221,23 +236,23 @@ public Response intercept(Chain chain) throws IOException {
         return chain.proceed(okReqBuilder.build());
     }
 
-    // Utility method to convert string to human-readable format
-    private String toHumanReadableAscii(String str) {
-        for (int i = 0, length = str.length(), c; i < length; i += Character.charCount(c)) {
-            c = str.codePointAt(i);
-            if (c > '\u001f' && c < '\u007f') {
-                continue;
-            }
-            Buffer buffer = new Buffer();
-            buffer.writeUtf8(str, 0, i);
-            for (int j = i; j < length; j += Character.charCount(c)) {
-                c = str.codePointAt(j);
-                if (c > '\u001f' && c < '\u007f') {
-                    buffer.writeUtf8CodePoint(c);
-                }
-            }
-            return buffer.readUtf8();
+    // Extracts query string parameters from a URL.
+    // Source: https://stackoverflow.com/questions/13592236/parse-a-uri-string-into-name-value-collection
+    @NonNull
+    private static Map<String, String> splitQuery(URL url) throws UnsupportedEncodingException {
+        Map<String, String> queryPairs = new LinkedHashMap<>();
+        String query = url.getQuery();
+        if (StringUtils.isBlank(query)) {
+            return Collections.emptyMap();
+        }
+        String[] pairs = query.split("&");
+        for (String pair : pairs) {
```

Comment:
I had some [similar code](https://github.com/aws-amplify/aws-sdk-android/blob/5dd0186faf85ad1b7482b641c3d69d70a31db465/aws-android-sdk-polly/src/test/java/com/amazonaws/services/polly/AmazonPollyPresigningClientTest.java#L217-L250) recently and @richardmcclellan raised [some good points](https://github.com/aws-amplify/aws-sdk-android/pull/2008#discussion_r453857894): what would happen here if the URL were malformed? Is a null pointer or out-of-range error thrown?